front/src/common/Map/Buttons/MapButtons.tsx

```diff
@@ -121,7 +133,17 @@ export default function MapButtons({
         <ButtonZoomOut zoomOut={() => zoomOut()} />
         <ButtonResetViewport updateLocalViewport={resetPitchBearing} bearing={bearing} />
         <ButtonMapSearch toggleMapSearch={() => toggleMapModal('SEARCH')} />
-        <ButtonMapSettings toggleMapSettings={() => toggleMapModal('SETTINGS')} />
+        {isInEditor && activeTool && toolState && setToolState ? (
+          <ButtonMapSettingsEditor
+            openModal={openModal}
+            editorState={editorState}
+            activeTool={activeTool}
+            toolState={toolState}
+            setToolState={setToolState}
+          />
+        ) : (
+          <ButtonMapSettings toggleMapSettings={() => toggleMapModal('SETTINGS')} />
+        )}
```

Comment:
You don't need to create a new component ButtonMapSettingsEditor, you can just change the toggleMapSettings function :)

You should create a function toggleMapSettings like this:
```
  const openMapSettingsModal = useCallBack(() => {
    if (isInEditor) {
      if (activeTool && toolState && setToolState) {
        openModal(
          <LayersModal
            initialLayers={editorState.editorLayers}
            frozenLayers={activeTool.requiredLayers}
            selection={
              activeTool.id === 'select-items'
                ? (toolState as unknown as SelectionState).selection
                : undefined
            }
            onChange={({ newLayers }) => {
              if (activeTool.id === 'select-items') {
                const currentState = toolState as unknown as SelectionState;
                setToolState({
                  ...currentState,
                  selection: currentState.selection.filter((entity) =>
                    EDITOAST_TO_LAYER_DICT[entity.objType as EditoastType].every((layer) =>
                      newLayers.has(layer)
                    )
                  ),
                } as SelectionState);
              }
            }}
          />,
          'lg'
        );
      }
    } else {
      toggleMapModal('SETTINGS')
    }
  }, [])
```
and then you can just 
```suggestion
        <ButtonMapSettings toggleMapSettings={openMapSettingsModal} />
        
```